src/query.ts

```diff
@@ -525,58 +572,86 @@ export function search(value: SourceArray, index: number): Lexer.Token {
   return Lexer.tokenize(value, start, index, expr, Lexer.TokenType.Search);
 }
 
-export function searchExpr(value: SourceArray, index: number): Lexer.Token {
-  const token = searchParenExpr(value, index) || searchTerm(value, index);
+export function searchExpr(
+  value: SourceArray,
+  index: number
+):
+  | Lexer.Token<Lexer.TokenType.SearchAndExpression>
+  | Lexer.Token<Lexer.TokenType.SearchOrExpression>
+  | Lexer.Token<Lexer.TokenType.SearchNotExpression>
+  | Lexer.Token<Lexer.TokenType.SearchParenExpression>
+  | Lexer.Token<Lexer.TokenType.SearchPhrase>
+  | Lexer.Token<Lexer.TokenType.SearchTerm>
+  | Lexer.Token<Lexer.TokenType.SearchWord> {
+  let token: TokenOfType<
+    | Lexer.TokenType.SearchAndExpression
+    | Lexer.TokenType.SearchOrExpression
+    | Lexer.TokenType.SearchNotExpression
+    | Lexer.TokenType.SearchParenExpression
+    | Lexer.TokenType.SearchPhrase
+    | Lexer.TokenType.SearchTerm
+    | Lexer.TokenType.SearchWord
+  > = searchParenExpr(value, index) || searchTerm(value, index);
```

Comment:
I like the fact we are being explict but is there any way we can alias some of these types to clean up the duplication

Reply:
Yes, I was planning to revisit this as a follow-in.  We should be able to reference `Token.SearchExpressionToken['value']` as the return type & type of the `token` variable here.  We can make a similar change throughout the codebase.  It just wasn't until making all of these changes & becoming familiar with the code that I recognized the ability to do that, and I just wanted to pause after getting everything initially typed. 😄